packages/std/src/ibcv2.rs

```diff
@@ -1,47 +1,61 @@
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
 
-use crate::{Binary, Timestamp};
+use crate::{Addr, Binary, Timestamp};
 
 /// Payload value should be encoded in a format defined by the channel version,
 /// and the module on the other side should know how to parse this.
 #[non_exhaustive]
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
-pub struct EurekaPayload {
+pub struct IBCv2Payload {
     /// The port id on the chain where the packet is sent to (external chain).
     pub destination_port: String,
     /// Version of the receiving contract.
     pub version: String,
-    /// Encoding used to serialize the [EurekaPayload::value].
+    /// Encoding used to serialize the [IBCv2Payload::value].
     pub encoding: String,
     /// Encoded payload data.
     pub value: Binary,
 }
 
-/// These are messages in the IBC lifecycle using the new Eureka approach. Only usable by IBC-enabled contracts
+/// These are messages in the IBC lifecycle using the new IBCv2 approach. Only usable by IBC-enabled contracts
 #[non_exhaustive]
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
-pub enum EurekaMsg {
+pub enum IBCv2Msg {
     /// Sends an IBC packet with given payloads over the existing channel.
     SendPacket {
         /// existing channel to send the tokens over
         channel_id: String,
         timeout: Timestamp,
-        payloads: Vec<EurekaPayload>,
+        payloads: Vec<IBCv2Payload>,
     },
 }
 
+/// The message that is passed into `ibcv2_packet_receive`
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[non_exhaustive]
+pub struct IBCv2PacketReceiveMsg {
+    pub payload: IBCv2Payload,
+    pub relayer: Addr,
```

Comment:
Don't we also need the source channel here? Otherwise you don't know where the payload is coming from.